tilewe/__init__.py

```diff
@@ -601,22 +623,31 @@ def on_tiles_filled(self, tiles: list[Tile]) -> None:
             del self.corners[r]
 
     def add_corner(self, tile: Tile) -> None:
-        if tile in self.corners or out_of_bounds(tile):
+        if tile in self.corners:
             return
 
-        bad: _PrpSet = 0
+        bad: _PrpSet = ~_PRP_SET_ALL
+
+        tgt = self._tgt
+        x, y = tile_to_coords(tile) 
 
         for rel in _PRP_REL_COORDS: 
-            pt = (rel[0] + tile[0], rel[1] + tile[1])
-            oob = out_of_bounds(pt)
-            if oob or self.board._tiles[pt] != 0:
+            pt = _REL_TILE_COORDS[rel]
+            pt = (pt[0] + x, pt[1] + y)
+            t = pt[0] + pt[1] * 20 
+            ib = in_bounds(pt)
+
+            if ib: 
+                if col := self._tiles[t]: 
+                    bad |= _PRP_WITH_REL_COORD[rel]
+                    if not (col - tgt): 
```

Comment:
looks so awful, but I can confirm for any lurkers that it improves performance :sob: